### PR TITLE
✅ Clean up types unit tests

### PR DESCRIPTION
**Pull request by alyssadsouza**
**Types**
- removes unnecessary unit tests that are just checking that Zod validation is working (we only need to test custom validation logic/regex)
- cleans up remaining unit tests to specify each test case with a description

### CODE REVIEW

File: packages/types/tests/entities/ClinicalProfile.spec.ts [hunk 1]
```diff
@@ -56,6 +58,8 @@ describe('ClinicalProfile', () => {
 				selfIdentifiedGender: 'Nulla labore amet minim nisi',
 			}).success,
 		).true;
+	});
+	it('Fails when gender is PREFER_NOT_TO_ANSWER and selfIdentifiedGender is NOT provided', () => {
```
Comment (samrichca) [inline packages/types/tests/entities/ClinicalProfile.spec.ts:62]: PREFER_NOT_TO_ANSWER -> PREFER_TO_SELF_IDENTIFY

File: packages/types/tests/entities/ClinicalProfile.spec.ts [hunk 2]
```diff
@@ -43,6 +43,8 @@ describe('ClinicalProfile', () => {
 				selfIdentifiedGender: undefined,
 			}).success,
 		).true;
+	});
+	it('Parses correctly when gender is PREFER_NOT_TO_ANSWER and selfIdentifiedGender is provided', () => {
```
Comment (samrichca) [inline packages/types/tests/entities/ClinicalProfile.spec.ts:47]: PREFER_NOT_TO_ANSWER -> PREFER_TO_SELF_IDENTIFY

File: packages/types/tests/entities/ClinicalProfile.spec.ts [hunk 3]
```diff
@@ -29,7 +29,7 @@ import {
 } from '../../src/entities/index.js';
 
 describe('ClinicalProfile', () => {
-	it('Must define self-identified gender if user chooses to self-identify', () => {
+	it('Parses correctly when gender is NOT PREFER_NOT_TO_ANSWER and selfIdentifiedGender is NOT provided', () => {
```
Comment (samrichca) [inline packages/types/tests/entities/ClinicalProfile.spec.ts:32]: can you change parses correctly to parses successfully? or something else about success

File: packages/types/tests/entities/ClinicalProfile.spec.ts (hunk 2 above)
Comment (alyssadsouza) [reply packages/types/tests/entities/ClinicalProfile.spec.ts:47]: my bad, thanks for catching that!

File: packages/types/tests/entities/ClinicalProfile.spec.ts (hunk 3 above)
Comment (samrichca) [inline packages/types/tests/entities/ClinicalProfile.spec.ts:32]: this one's OK

**Review by anncatton — COMMENTED** [review]
I think we can remove the tests for `OhipNumber`, `Name` and `PhoneNumber`, they are just testing zod parse i think?

But maybe keep the uppercase transform test for `PostalCode`, i'd say that's a bit more custom. @mistryrn does that sound sensible?

**Review by anncatton — APPROVED** [review]
Discussed my previous comment on slack, i am appeased 😛 Approved!

**Pull request merged**